mixer/pkg/adapter/test/integration.go

```diff
@@ -0,0 +1,270 @@
+package test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"path"
+	"path/filepath"
+	"reflect"
+	"runtime"
+	"testing"
+
+	"google.golang.org/grpc"
+
+	istio_mixer_v1 "istio.io/api/mixer/v1"
+	"istio.io/istio/mixer/pkg/adapter"
+	"istio.io/istio/mixer/pkg/attribute"
+	"istio.io/istio/mixer/pkg/config/storetest"
+	"istio.io/istio/mixer/pkg/server"
+	"istio.io/istio/mixer/pkg/template"
+)
+
+type (
+	// TestCase fully defines an adapter integration test
+	TestCase struct {
+		// Cfgs is a list of CRDs that Mixer will read.
+		Cfgs []string
+		// ParallelCalls is a list of test calls to be made to Mixer
+		// in parallel.
+		ParallelCalls []Call
+		// Expected json string for a WantObject object.
+		//
+		// New test can start of with an empty "{}" string and then
+		// get the baseline from the failure logs upon execution.
```

Comment:
The "run and see what you get" strategy is ok here, but a high level of what this should look like would help. For example: This is compared to a json serialization of a Result struct, where Adapter State is what getState returns.